NDTensors/src/lib/Sectors/src/category_definitions.jl

```diff
@@ -0,0 +1,115 @@
+
+#
+# 2D rotation group U(1)
+#
+
+U(N::Int) = Category("U", N)
+
+fusion_rule(::CategoryType"U", n1, n2) = [(n1 + n2)]
```

Comment:
I feel like we already discussed this kind of design, but what makes sense to me is something like:
```julia
struct U
  N::Int
end
CategoryName(c::U) = CategoryName{String7("U")}()
Category(c::U) = Category(CategoryName(c), c.N)
category(l::Label{<:Any,CategoryName"U"}) = U(only(values(l))) # Not sure if this is right, and ideally the syntax would be simplified

fusion_rule(c1::U, c2::U) = [(c1.N + c2.N)]
```
i.e. the user defines a symmetry/category object and fusion rules for that object, and then defines conversion back and forth between their type and the "canonical" `Category`/`Label` types.

Reply:
One reason for going in that direction would be that `U` could in principle just be used directly as a sector of a `GradedAxes`, instead the `Label`/`Sector` type. So we can think of the job of the `Sectors` module as providing canonical `Label`/`Sector` types which you can convert a custom symmetry type to in order to get fancier behavior in terms of mixing symmetry types together.

Reply:
Yes, we can definitely add back having the separate types for each group or category. (So an actual U type, SU type, etc.) 

One thing I'm missing about the above, though, is that in your example it seems the `N` field is storing the actual value of the label e.g. the number of particles. Whereas I was envisioning the Category types as just specifying which group the values go with. We could of course make a type that holds the group name and values together, but that's what `Label` already was supposed to be. 

But I'll think about it some more, because it could be a good direction. So like we could have something like:
```julia
struct U1 <: AbstractSector
  n::int
end

struct SU{N} where {N} <: AbstractSector
  J::NTuple{Half{Int},N-1}
end
```
i.e. distinct types that both specify a group and the particular irrep (or "sector value") of that group.  That would certainly be nice from the perspective of defining fusion rules etc. directly as dispatching on these types and returning arrays of such objects.

Then we could reimagine `Label` as being just a means to store the above structs in a common, dynamic, format with an optional label name. And then Sector as a sorted set of those.

Reply:
Probably should schedule a chat about this since it's more fundamental about the whole design. It could be better, but it's a big change so would need to be on the same page about it.

Reply:
Yeah it's a good question if these objects should be just the group or an irrep of the group. It seems better both for defining fusion rules and conversions for them to be irreps but I haven't thought through the implications of the design beyond that. Sure, let's set up a chat about it.

Reply:
It really could be a better design, since for one thing it would push off the issue of the `Label` type having to hold various value types (e.g. tuple of different lengths) inside that could then "collide" when they are to be stored in a single `Sector`. So maybe a design where the categories and values are more designed together could clarify how to do these conversions/promotions more nicely. (Like finding a common type when two of these custom types "meet" each other.)